scanpy/plotting/_tools/scatterplots.py

```diff
@@ -1181,6 +1166,32 @@ def _check_scale_factor(
         return 1.0
 
 
+def _check_spatial_data(
+    uns: Mapping, library_id: Union[Empty, None, str]
+) -> Tuple[Optional[str], Optional[Mapping]]:
+    """
+    Given a mapping, try and extract a library id/ mapping with spatial data.
+
+    Assumes this is `.uns` from how we parse visium data.
+    """
+    spatial_mapping = uns.get("spatial", {})
```

Comment:
I like this, however, in spatial data without image we don't even have `adata.uns[["spatial"]` so this wil break